### ISSUE TITLE
Option to fail if property type is ambiguous

### ISSUE THREAD

**Issue by balhoff** [owlcs/owlapi#734]
We periodically run into hard-to-diagnose issues that stem from properties in RDF files being interpreted as annotation properties rather than object properties. This is usually the result of some step in an ontology processing pipeline that fails to load a required imported ontology. The OWL API defaults to interpreting ambiguous properties as annotation properties.

Could there be an option (I guess in `OWLOntologyLoaderConfiguration`) to treat ambiguous properties as an error? Does the RDF mapping spec say anything about this?

I tried setting `isStrict` to `true`, but it didn't seem to alter this behavior. (It's not clear what `isStrict` is for).

**Comment by cmungall** [comment]
In theory failing to load an import would throw an error, but in the past there have been cases where some kind of external I/O error resulted in what the owlapi thinks is a valid ontology, e.g. #657. I'm not sure it's possible to eliminate all possible scenarios. So this will continue to be an issue.

Even if I/O is always successful, we operate in an environment where we are dependent on external ontologies. Someone in an upstream ontology might do something like obsolete an object property that would silently change the semantics of the importing ontology (of course this is true in the general case, but pragmatically there is a big difference in forcing a syntactic change vs a semantic one).

TL;DR - I support an option to enforce strictness.

**Comment by ignazio1977** [comment]
> I tried setting isStrict to true, but it didn't seem to alter this behavior. (It's not clear what isStrict is for).

Can you share how you set this? It is supposed to switch to strict parsing, where untyped properties would throw an exception

**Comment by balhoff** [comment]
Testing in an interactive Scala console with OWL API 4.5.1:

```scala
OWLManager.createOWLOntologyManager().loadOntologyFromOntologyDocument(new FileDocumentSource(new File("test.ttl")), new OWLOntologyLoaderConfiguration().setStrict(true))
// res11: org.semanticweb.owlapi.model.OWLOntology = Ontology(OntologyID(Anonymous-23)) [Axioms: 3 Logical Axioms: 0] First 20 axioms: {Declaration(NamedIndividual(<http://example.org/test#my_finger>)) Declaration(NamedIndividual(<http://example.org/test#my_hand>)) AnnotationAssertion(<http://purl.obolibrary.org/obo/BFO_0000050> <http://example.org/test#my_finger> <http://example.org/test#my_hand>) }
```

test.ttl:

```turtle
@prefix : <http://example.org/test#> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix xml: <http://www.w3.org/XML/1998/namespace> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@base <http://example.org/test> .


#################################################################
#    Individuals
#################################################################

###  http://example.org/test#my_finger
:my_finger rdf:type owl:NamedIndividual ;
           <http://purl.obolibrary.org/obo/BFO_0000050> :my_hand .


###  http://example.org/test#my_hand
:my_hand rdf:type owl:NamedIndividual .


###  Generated by the OWL API (version 4.2.6.20160910-2108) https://github.com/owlcs/owlapi
```
